packages/apputils/src/sessioncontext.tsx

```diff
@@ -1265,14 +1304,26 @@ export namespace SessionContext {
    * Get the default kernel name given select options.
    */
   export function getDefaultKernel(options: IKernelSearch): string | null {
+    const { preference } = options;
+    const { shouldStart } = preference;
+
+    if (shouldStart === false) {
+      return null;
+    }
+
     return Private.getDefaultKernel(options);
   }
 }
 
 /**
  * The default implementation of the client session dialog provider.
  */
-export const sessionContextDialogs: ISessionContext.IDialogs = {
+export class SessionContextDialogs implements ISessionContext.IDialogs {
+  constructor(settings?: ISettingRegistry.ISettings, translator?: ITranslator) {
+    this._settings = settings || null;
```

Comment:
SessionContextDialogs works with `null` settings. This makes preparing testing environment a lot easier.